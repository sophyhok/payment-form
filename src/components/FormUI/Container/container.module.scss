.container {
  display: grid;

  &__fontColor_success {
    color: #00FF00;
  }

  &__fontColor_danger {
    color: #FF0000;
  }

  &__fontColor_dark {
    color: #1D2A3C;
  }

  &__fontColor_warning {
    color: #F89C24;
  }

  &__gap_none {
    gap: 0 0;
  }
  &__gap_xs {
    gap: 12px 12px;
  }
  &__gap_sm {
    gap: 16px 16px;
  }
  &__gap_md {
    gap: 20px 20px;
  }
  &__gap_lg {
    gap: 24px 24px;
  }

  &__align_start {
    align-items: start;
  }
  &__align_end {
    align-items: end;
  }
  &__align_center {
    align-items: center;
  }
  &__align_stretch {
    align-items: stretch;
  }

  &__justify_start {
    justify-items: start;
  }
  &__justify_end {
    justify-items: end;
  }
  &__justify_center {
    justify-items: center;
  }
  &__justify_stretch {
    justify-items: stretch;
  }
}