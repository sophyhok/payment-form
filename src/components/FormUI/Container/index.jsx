import React from 'react';
import { string, oneOf, node } from 'prop-types';
import classnames from 'classnames';

import styles from './container.module.scss';

const propTypes = {
  children: node,
  align: oneOf(['start', 'end', 'center', 'stretch']),
  justify: oneOf(['start', 'end', 'center', 'stretch']),
  gap: oneOf(['none', 'xs', 'sm', 'md', 'lg']),
  fontColor: oneOf(['danger', 'dark', 'success', 'warning']),
  testId: string,
}

const defaultProps = {
  children: null,
  align: 'start',
  gap: 'sm',
  justify: 'start',
  fontColor: 'dark',
  testId: 'container'
}

const Container = ({align, children, justify, gap, fontColor, testId}) => {
  return (
    <div className={classnames(styles.container,
      styles[`container__align_${align}`],
      styles[`container__justify_${justify}`],
      styles[`container__gap_${gap}`],
      styles[`container__fontColor_${fontColor}`]
      )}
      data-testid={testId}
    >
      {children}
    </div>
  )
}

Container.propTypes = propTypes;
Container.defaultProps = defaultProps;

export default Container;