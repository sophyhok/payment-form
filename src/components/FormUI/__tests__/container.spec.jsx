import React from 'react';
import { render } from '@testing-library/react';

import Container from '../Container';

let container;

it('should match snapshot', () => {
  const { asFragment } = render(<Container />);
  expect(asFragment()).toMatchSnapshot();
});

it('should render fontColor danger when props is passed', () => {
  const { getByTestId } = render(<Container fontColor="danger" />);
  container = getByTestId('container');
  expect(container.className).toMatch('container__fontColor_danger');
});

it('should render gap sm when no props is passed', () => {
  const { getByTestId } = render(<Container />);
  container = getByTestId('container');
  expect(container.className).toMatch('container__gap_sm');
});
