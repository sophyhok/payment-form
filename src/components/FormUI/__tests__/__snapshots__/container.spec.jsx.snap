// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="container container__align_start container__justify_start container__gap_sm container__fontColor_dark"
    data-testid="container"
  />
</DocumentFragment>
`;
